### PR TITLE
[IDE] Migrate TypeContextInfo to solver-based

### PR DESCRIPTION
**Pull request by ahoppen**

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

File: lib/IDE/TypeContextInfo.cpp [hunk 1]
```diff
@@ -85,19 +87,42 @@ void ContextInfoCallbacks::completeCaseStmtBeginning(CodeCompletionExpr *E) {
...
+};
+
 void ContextInfoCallbacks::doneParsing(SourceFile *SrcFile) {
   if (!ParsedExpr)
     return;
 
-  typeCheckContextAt(TypeCheckASTNodeAtLocContext::declContext(CurDeclContext),
-                     ParsedExpr->getLoc());
-
-  ExprContextInfo Info(CurDeclContext, ParsedExpr);
```
Comment (rintaro) [inline lib/IDE/TypeContextInfo.cpp:95]: Could we remove `#include "ExprContextAnalysis.h"`?

**Review by rintaro — APPROVED** [review]

File: lib/IDE/TypeContextInfo.cpp (hunk 1 above)
Comment (ahoppen) [reply lib/IDE/TypeContextInfo.cpp:95]: We still need it for
```cpp
void typeCheckContextAt(TypeCheckASTNodeAtLocContext TypeCheckCtx,
                        SourceLoc Loc);
```

At some point, we should probably just move that function somewhere else.

**Pull request merged**